app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden is False:
+            self.hidden = True
+        elif self.hidden is True:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(other: Animal) -> None:
+        if isinstance(other, Herbivore) \
+                and other.hidden is False:
```

Comment:
Replace other.hidden is False  condition with a more straightforward 'not' check and try to avoid using backslash